stdlib/toolchain/Compatibility50/BuiltinProtocolConformances.cpp

```diff
@@ -0,0 +1,98 @@
+//===--- BuiltinProtocolWitnessTables.cpp ---------------------------------===//
```

Comment:
Any time we link against `Compatibility50`, we should also link against `Compatibility51`, since we'd be backward deploying to support both 5.0 and 5.1 runtimes at that point. Instead of duplicating this code in both compatibility libraries, I think you should be able to put it once in the `Compatibility51` library, and refer to the `51` symbols from within `50`'s hooks table.

Reply:
Oh that's interesting. I'll try and see if I can get this working.

Reply:
> refer to the `51` symbols from within `50`'s hooks table.

It seems to me that this is referring to replacing the `swift50override_conformsToProtocol` in `50`'s hook table with `51`'s, but that doesn't seem right because it doesn't use any of the backported bug fixes for `50`. I assume I'm interpreting this incorrectly, but does it make sense to call `51`'s implementation within `50`'s implementation?

Reply:
Yeah, in the case of conformsToProtocol where there are 5.0-specific fixes, we should have the swift50override perform those fixes then chain into the 5.1 version.